### PR TITLE
Feat/seed

### PR DESCRIPTION
**Pull request by caleb-bitaccess**
#  Add seed and derivationPath to all UHdConfig

To allow all coins have a unified config interface

- add UHd*Payments classes
- add UHd*PaymentsConfig
- add tests

### CODE REVIEW

**Review by dylanseago — CHANGES_REQUESTED** [review]
I would recommend moving duplicate tests into helpers within each test folder. Having to maintain all these additional ones will make future changes more tedious

File: packages/coinlib-bitcoin-cash/src/UHdBitcoinCashPayments.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+export class UHdBitcoinCashPayments extends HdBitcoinCashPayments {
+  readonly seed: string | null
+  constructor(config: UHdBitcoinCashPaymentsConfig) {
+    assertType(UHdBitcoinCashPaymentsConfig, config)
+    let hdKey: string
+    let seed: string | null = null
+    if (SeedBitcoinCashPaymentsConfig.is(config)) {
+      seed = config.seed
+      const rootNode = determineHdNode(seed)
```
[the remark this answers is not in the document]
Comment (caleb-bitaccess) [reply packages/coinlib-bitcoin-cash/src/UHdBitcoinCashPayments.ts:20]: `derivationPath` is passed in config to parent Hd class.  Need to change the rest coins to support a customized derivation path in future sprints.

File: packages/coinlib-ripple/src/types.ts [hunk 2]
```diff
@@ -93,7 +117,7 @@ export const AccountRipplePaymentsConfig = extendCodec(
 export type AccountRipplePaymentsConfig = t.TypeOf<typeof AccountRipplePaymentsConfig>
 
 export const RipplePaymentsConfig = t.union(
-  [HdRipplePaymentsConfig, AccountRipplePaymentsConfig],
+  [HdRipplePaymentsConfig, UHdRipplePaymentsConfig, AccountRipplePaymentsConfig],
```
[the remark this answers is not in the document]
Comment (caleb-bitaccess) [reply packages/coinlib-ripple/src/types.ts:120]: Nope. This trick does not help. TS still thinks the type is too complex.

File: packages/coinlib-litecoin/test/UHdLitecoinPayments.test.ts [hunk 3]
```diff
@@ -89,125 +78,125 @@ describe('UHdLitecoinPayments', () => {
...
-      toAddress: addresses[2],
-      toIndex: null,
-      toExtraId: undefined,
-      toPayport: { address: addresses[2] },
-    })
-  })
-  it('resolveFromTo throws for address as from', async () => {
-    await expect(payments.resolveFromTo(EXTERNAL_ADDRESS as any, 0)).rejects.toThrow()
-  })
+// function runHardcodedPublicKeyTests(
```
Comment (dylanseago) [inline packages/coinlib-litecoin/test/UHdLitecoinPayments.test.ts:81]: What's with these comments?

**Review by dylanseago — APPROVED** [review]

**Pull request merged**